File: Problems/0383-ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        cnt = [0] * 26

        for c in magazine:
            cnt[ord(c) - 97] += 1

        for c in ransomNote:
            idx = ord(c) - 97
            cnt[idx] -= 1
            if cnt[idx] < 0:
                return False

        return True

File: Problems/0383-ransom-note.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int cnt[26] = {};

        for (char c : magazine)
            cnt[c - 'a']++;

        for (char c : ransomNote) {
            int idx = c - 'a';

            if (--cnt[idx] < 0)
                return false;
        }

        return true;
    }
};